forem/forem
# Split Settings::Campaign from SiteConfig

## What type of PR is this? (check all applicable)


## Description

This PR extracts campaign settings from `SiteConfig` into `Settings::Campaign` and changes all related code.

## Related Tickets & Documents

https://github.com/forem/rfcs/pull/126

## QA Instructions, Screenshots, Recordings

* run `bin/setup` (or `rails data_updates:run`) and check if the data has been correctly migrated
* Go to `/admin/config` and make sure the campaign settings behave as expected.

### UI accessibility concerns?

None, as no user-facing behavior was changed, only backend implementation details.

## Added tests?

- [x]  Yes, added/updated as appropriate

## [Forem core team only] How will this change be communicated?

- [X] This change does not need to be communicated, and this is why not: this is purely a backend refactoring without user-facing changes

## [optional] Are there any post deployment tasks we need to perform?

- [ ] Keep an eye on our error rate to ensure this works as expected
